### PR TITLE
add feature test and functionality for adding items to cart

### PR DESCRIPTION
**Pull request by blackknight75**
create add to cart button
add flash message for adding items to cart
add increment to flash message of cart
add method to cart model
add method to count all items in cart
add tests for cart being incremented, all tests pass.
add cart to application controller
remove the instance variables of cart from other controllers

### CODE REVIEW

File: app/controllers/carts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+include ActionView::Helpers::TextHelper
+
+  def create
+    item = Item.find(params[:item_id])
+
+    @cart.add_item(item.id)
+    session[:cart] = @cart.contents
+
+    flash[:notice] = "You have added #{pluralize(@cart.count_of(item.id), item.title)} to cart"
```
Comment (scmountain) [inline app/controllers/carts_controller.rb]: clean this up at the comma

File: config/routes.rb [hunk 2]
```diff
@@ -6,6 +6,8 @@
   get '/items', to: 'items#index'
 
   resources :items, only: [:new]
+
+  resources :carts
   # resources :categories, controller: 'categories', param: :slug
```
Comment (scmountain) [inline config/routes.rb]: lets delete all uncommented lines out that we dont need.

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -1,5 +1,11 @@
 class ApplicationController < ActionController::Base
+
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+  before_action :set_cart
```
Comment (blackknight75) [inline app/controllers/application_controller.rb]: Nice work with the before action. Saved us some redundancy in our controller. @scmountain @brandonrandall

File: app/models/cart.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
+class Cart
+attr_reader :contents
+
+  def initialize(initial_contents)
+    @contents = initial_contents || {}
+  end
+
+  def total_count
+    @contents.values.sum
```
Comment (blackknight75) [inline app/models/cart.rb]: Does this need to be an instance variable?

File: config/routes.rb (hunk 2 above)
Comment (blackknight75) [inline config/routes.rb]: That sounds logical.

**Pull request merged**